server/loops.lua

```diff
@@ -1,16 +1,18 @@
 local config = require 'config.server'
 
 local function removeHungerAndThirst(src, player)
-    local newHunger = player.PlayerData.metadata.hunger - config.player.hungerRate
-    local newThirst = player.PlayerData.metadata.thirst - config.player.thirstRate
+    local playerState = Player(src).state
+    if not playerState.isLoggedIn then return end
+    local newHunger = playerState.hunger - config.player.hungerRate
+    local newThirst = playerState.thirst - config.player.thirstRate
     if newHunger <= 0 then
         newHunger = 0
     end
     if newThirst <= 0 then
         newThirst = 0
     end
-    player.Functions.SetMetaData('thirst', newThirst)
-    player.Functions.SetMetaData('hunger', newHunger)
+    playerState:set('hunger', newHunger, true)
+    playerState:set('thirst', newThirst, true)
```

Comment:
we should use math.min here instead of wierd if statements to check if the value is less than 0

```lua
playerState:set('hunger', math.min(0, newHunger), true)
playerState:set('thirst', math.min(0, newThirst), true)
```